### PR TITLE
Add note that PUT requests require the BodyParamsMiddleware

### LINKED ISSUE #17: [Question] $request->getParsedBody() returns empty array  using PUT/PATCH methods
I'm having difficulty in parsing the $request->getParsedBody() for PUT/PATCH method - it only returns empty array.

I've enabled Body Parsing Middleware on my controller and it works with POST method.

https://docs.zendframework.com/zend-expressive/features/helpers/body-parse/

I'm trying to implement the REST Controller by Alejandro Celaya

https://blog.alejandrocelaya.com/2016/06/24/dispatch-rest-like-requests-with-a-single-controller-class-in-zend-expressive/

Appreciate your input here. Thanks in advance.

---

Originally posted by @gabbydgab at https://github.com/zendframework/zend-expressive/issues/371

### PR DESCRIPTION
**Pull request by settermjd**
|    Q          |   A
|-------------- | ------
| Documentation | **yes**
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

While building a small application that includes PUT routes, I came across the need for including `BodyParamsMiddleware` in the route's request pipeline, otherwise calls to `getParsedBody` return an empty array. Given that I didn't find anything about this, specifically, in the docs for the middleware, I created this small change to include it. I also created it to help resolve https://github.com/mezzio/mezzio/issues/17.

### CODE REVIEW

File: docs/book/v3/features/helpers/body-parse.md [hunk 1]
```diff
@@ -13,6 +13,9 @@ By default, this middleware will detect the following content types:
   uploads)
 - `application/json`, `application/*+json` (JSON payloads)
 
+> ### This middleware **must** be used for PUT requests
```
Comment (froschdesign) [inline docs/book/v3/features/helpers/body-parse.md:16]:
```suggestion
> ### This Middleware **Must** Be Used for PUT Requests
>
```

Comment (settermjd) [reply docs/book/v3/features/helpers/body-parse.md:16]: Thanks for spotting this, Frank.

File: docs/book/v3/features/helpers/body-parse.md [hunk 2]
```diff
@@ -13,6 +13,9 @@ By default, this middleware will detect the following content types:
...
+> If it is not a part of the middleware pipeline for routes using PUT requests, then calls to `ServerRequestInterface::getParsedBody()` will return an empty array—even if the request included a body.
```
Comment (froschdesign) [inline docs/book/v3/features/helpers/body-parse.md:17]: I had read something else at first: "array-even" therefore my suggestion:
```suggestion
> If it is not a part of the middleware pipeline for routes using PUT requests, then calls to `ServerRequestInterface::getParsedBody()` will return an empty array — even if the request included a body.
```

**Review by froschdesign — APPROVED** [review]

**Comment by boesing** [comment]
I've retargeted to 3.6.x as 4.0 may take a while

**Pull request merged**